c/tskit/trees.c

```diff
@@ -7007,10 +7002,10 @@ tsk_treeseq_kc_distance(const tsk_treeseq_t *self, const tsk_treeseq_t *other,
     if (ret != 0) {
         goto out;
     }
-    tsk_tree_position_next(&tree_pos[0]);
-    tsk_bug_assert(tree_pos[0].index == 0);
+    tsk_tree_position_next(&trees[0].tree_pos);
```

Comment:
Ah no, that's not the point - we need to use the *internal* calls to tree_pos_next. Calling externally like this should break all sorts of things.